### PR TITLE
Use nar to create a self-contained exectuable

### PR DESCRIPTION
**Pull request by SimenB**
Blocked on h2non/nar#140

/cc @ivarconr

### CODE REVIEW

**Comment by SimenB** [comment]
This thing should allow us to download it after i actually works:

```js
const { request } = require('https');
const { exec } = require('child_process');

const resultingFile = 'cdn-uploader.nar';

function makeRequest ({ host, path }, onEnd) {
    const options = {
        host,
        path,
        headers: {
            'user-agent': 'My-Cool-GitHub-App' // GitHub is happy with a unique user agent
        },
    };

    function callback (response) {
        let str = '';

        response.on('data', function (chunk) {
            str += chunk;
        });

        response.on('end', () => {
            onEnd(str);
        });
    }

    request(options, callback).end();
}

makeRequest({ host: 'api.github.com', path: '/repos/finn-no/cdn-uploader/releases/latest' }, releaseLookup => {
    const res = JSON.parse(releaseLookup);
    const downloadUrl = res.assets[0].browser_download_url;

    console.log(`Downloading ${downloadUrl}`);

    exec(`curl -Lo ${resultingFile} ${downloadUrl}`, err => {
        if (err) {
            throw err;
        }

        console.log(`Successfully downloaded ${resultingFile}`);

        exec(`chmod 777 ${resultingFile}`, err => {
            if (err) {
                throw err;
            }

            console.log(`${resultingFile} is made executable`);
        })
    });
});
```

**Comment by SimenB** [comment]
Kinda forgot about this. Current status is that the bundling works, but the file has to be extracted before use, which takes some time. It would be preferable if we could pre-extract it, but I can't get it to work.
Low pri at the moment.

**Comment by SimenB** [comment]
@ivarconr @sveisvei @trygve-lie Ok, this works now. Travis should also automatically upload new versions to GH releases whenever we tag, but that's untested

File: .travis.yml [hunk 1]
```diff
@@ -1,14 +1,22 @@
 language: node_js
 node_js:
-  - 7
-  - 6
+- 7
```
Comment (SimenB) [inline .travis.yml]: travis cli made the whitespace changes, so I'm keeping them in

File: .travis.yml [hunk 2]
```diff
@@ -1,14 +1,23 @@
...
 env:
-  - CXX=g++-4.8
+- CXX=g++-4.8
 addons:
   apt:
     sources:
     - ubuntu-toolchain-r-test
     packages:
     - g++-4.8
+deploy:
```
Comment (SimenB) [inline .travis.yml]: this section is generated by travis cli, see https://docs.travis-ci.com/user/deployment/releases/ for docs

**Comment by SimenB** [comment]
We still have to figure out the best way to get it into finnbuild. My script above works, but there might be more elegant solutions? Some combination of curl + jq + curl might work fine

**Comment by SimenB** [comment]
Shellscript for it

```sh
#!/usr/bin/env sh

set -e

releases_url=https://api.github.com/repos/finn-no/cdn-uploader/releases/latest
resulting_file=cdn-uploader.nar
jq_selector=".assets[0].browser_download_url"

download_url=$(curl "$releases_url" --silent | jq "$jq_selector" | sed 's/\"//g')

echo Downloading "$download_url"...

curl -Lo "$resulting_file" "$download_url" --silent

chmod +x "$resulting_file"

echo "$resulting_file" succesfully downloaded!
```

**Comment by ivarconr** [comment]
Awesome! lgtm! :white_check_mark:

File: .travis.yml [hunk 3]
```diff
@@ -1,14 +1,23 @@
...
+  provider: releases
+  api_key:
+    secure: Z2QwMzGV8NuhX1Ctt7MEc5TCloo0gv22CEGarIFVZUUwocGZ5sqs3EgpHwa04qezo4UfoGBGuRpsuBA+MMX6/TD1E9fYJp9O2g/yRSOtW2AVPwKU1Ue3tIXVZm4mqAc6EKDiuYviDQlwHGbYMs2HW8wngiPg2Zv8glVRKKOypwcOFnbfmOZmsbkWZ32a0+SSDUanq3raQ50HEYiyCBGfr7+XRizy37y8b5pu3a/euyOE1SxnDcbofcAbmVMBSh8k67etQfTccdOzkyuR2IusFoFgY1kW1CWztqjdvutelkmVMcExij9isXi6BvSRWQ+qss6xjZvzKNVdIQ5gWOyb2+QZ5VjnXrAckZYA9eS2f8kGepnhk7nVE54ya61rPwVdS9qg6mCD99vE+rBKxaPSPht50P4LfqKg2M8N1mc4ghC80H6cvc20rmmtpNIe08ADh1lT8eNAA1XnNxU7tQXZwpVAqrURIDmLbsVwGRoWfF4+Ah+DT4onLp5PhwDhVZ4F5jUchTjuInRnAcTk8l2IUffwv8TTFOC5nqFTg2vG/pMH/BG1PpoB8fI2YnmQHygL3k528yXEHgLYMLuZTmQ3+pY9UrIBQCkNmr3ie7skTnTRZrCQqVJNPNssV2xpsgWtc+wYnzBqp2XTs7U87TIcSxD/4tQZb7IUZ8zA7dsIvy4=
```
Comment (ivarconr) [inline .travis.yml]: what it this api_key?

Comment (SimenB) [inline .travis.yml]: encrypted username and pw, for this repo specifically

**Pull request merged**